benchmark/single-source/ArrayRRC.swift

```diff
@@ -0,0 +1,113 @@
+//===--- ArrayRRC.swift ----------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import TestsUtils
+import Darwin
+
+fileprivate class DummyObject {}
+
+fileprivate func makeRefcounted(count: Int) -> [DummyObject] {
+  var result = (0 ..< count - 1).map { _ in DummyObject() }
+  blackHole(result)
+  result.append(DummyObject()) //force a resize if result was perfectly fitted
+  return result
+}
+
+fileprivate func ranges(count: Int) -> [(Range<Int>, String)] {
+  var results:[(Range<Int>, String)] = []
+  results.append((0 ..< (count / 3), "head"))
+  results.append((((count / 3) * 2) ..< count, "tail"))
+  return results
+}
+
+fileprivate let sourceSizes = [(10_000, "lg"), (99, "sm")/*, (1, "xs")*/]
+fileprivate let destSizes = [(10_000, "lg"), (99, "sm")]
+
+
+// We're currently only testing non-uniquely-referenced Arrays of
+// refcounted objects to cut down on the excessive number of subtests.
+// Revert 3362243cd6f1b6c16dc8d10dd27c8f3e881ecbc7 if you want the full thing
+fileprivate func configs() -> [BenchmarkInfo] {
+  var configs: [BenchmarkInfo] = []
+  for (sourceCount, sourceName) in sourceSizes {
```

Comment:
Please do not add this massive amount of benchmarks. Adding so many benchmarks for each compiler feature/change does not scale.
I suggest to either
* add a single benchmark which runs all the different cases and results in a "mean" value for all that cases.
* pick one or a few representative benchmarks and add the `.skip` tag to all others. The "skipped" benchmarks can then be run manually by adding the `--skip-tags` option. Or add a new tag case value which is skipped by default.